app/signup/page.tsx

```diff
@@ -0,0 +1,58 @@
+"use client";
+
+import { useSession } from "next-auth/react";
+import { redirect } from "next/navigation";
+import { useDict } from "../utils/useDictHook";
+import { useState } from "react";
+import { Form, IFormInput } from "../components/form";
+import { SubmitHandler } from "react-hook-form";
+import { signIn } from "next-auth/react";
+import { createUserWithEmailAndPassword } from "firebase/auth";
+import { MdErrorOutline } from "react-icons/md";
+import { auth } from "@/firebase";
+import Link from "next/link";
+
+const Page = () => {
+  const { status } = useSession();
+  if (status === "authenticated") redirect("/main");
+  const [error, setError] = useState("");
+  const dict = useDict();
+  const signup: SubmitHandler<IFormInput> = async (data: {
+    email: string;
+    password: string;
+  }) => {
+    setError("");
+    const email = data.email;
+    const password = data.password;
+
+    await createUserWithEmailAndPassword(auth, email, password)
```

Comment:
I would suggest do not mix an async/await and callbacks. Choose one approach to avoid a confusion.